core/db.py

```
@@ -106,8 +139,9 @@ def execute(self, statement: Any, *args: Any, **kwargs: Any):
             result = conn.execute(stmt, *args, **kwargs)
             try:
                 conn.commit()
-            except Exception:
-                # Not all statements require/allow commit; ignore commit errors
+            except Exception:  # nosec B110
+                # Some statements (DDL, read-only queries) don't require commit
+                # For those cases, the exception is expected and can be safely ignored
                 pass
```

Ruff (0.13.3)

`try`-`except`-`pass` detected, consider logging the exception

(S110)